[interop] Swift enum with more than one case and no associated values crash the compiler

Repro steps:

Modify `test/Interop/SwiftToCxx/enums/swift-enum-implementation.swift` to contain only the following:
```swift
public enum E {
    case foobar
    case baz
}
```
Modify `test/Interop/SwiftToCxx/enums/swift-enum-implementation-execution.cpp` to contain only the following:
```c++
int main() {
    #include <cassert>
    #include "enums.h"
    
    using namespace Enums;

    auto e = E::foobar();
    return  0;
}
```

Running `swift-enum-implementation-execution.cpp`, the test will crash with the following error:

```
/Users/augusto/Developer/swift/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-arm64/test-macosx-arm64/Interop/SwiftToCxx/enums/Output/swift-enum-implementation-execution.cpp.tmp/enums.h:676:39: error: no type named 'Hasher' in namespace 'swift'
  SWIFT_INLINE_THUNK void hash(swift::Hasher& hasher) const SWIFT_SYMBOL("s:5Enums1EO4hash4intoys6HasherVz_tF");
                               ~~~~~~~^
/Users/augusto/Developer/swift/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-arm64/test-macosx-arm64/Interop/SwiftToCxx/enums/Output/swift-enum-implementation-execution.cpp.tmp/enums.h:910:42: error: no type named 'Hasher' in namespace 'swift'
  SWIFT_INLINE_THUNK void E::hash(swift::Hasher& hasher) const {
                                  ~~~~~~~^
/Users/augusto/Developer/swift/build/Ninja-RelWithDebInfoAssert+stdlib-RelWithDebInfo/swift-macosx-arm64/test-macosx-arm64/Interop/SwiftToCxx/enums/Output/swift-enum-implementation-execution.cpp.tmp/enums.h:911:67: error: no member named '_impl_Hasher' in namespace 'swift::_impl'
  return _impl::$s5Enums1EO4hash4intoys6HasherVz_tF(swift::_impl::_impl_Hasher::getOpaquePointer(hasher), _impl::swift_interop_passDirect_Enums_uint8_t_0_1(_getOpaquePointer()));
```

Version:
```
Swift version 5.9-dev (LLVM 11017a372a2fcf3, Swift 57128dd0ee11e00)
```

--- a/lib/PrintAsClang/DeclAndTypePrinter.cpp
+++ b/lib/PrintAsClang/DeclAndTypePrinter.cpp
@@ -2784,8 +2784,12 @@ static bool isExposedToThisModule(const ModuleDecl &M, const ValueDecl *VD,
   if (VD->hasClangNode())
     return true;
   auto *mc = VD->getModuleContext();
-  if (mc == &M || mc->isStdlibModule())
+  if (mc == &M)
     return true;
+  // Only certain declarations are exposed from
+  // the standard library.
+  if (mc->isStdlibModule())
+    return hasExposeAttr(VD);
   return exposedModules.count(mc->getName().str());
 }
 

--- a/test/Interop/SwiftToCxx/enums/swift-enum-implementation.swift
+++ b/test/Interop/SwiftToCxx/enums/swift-enum-implementation.swift
@@ -4,6 +4,9 @@
 
 // RUN: %check-interop-cxx-header-in-clang(%t/enums.h -Wno-unused-private-field -Wno-unused-function)
 
+// RUN: %target-swift-frontend %s -typecheck -module-name Enums -enable-experimental-cxx-interop -emit-clang-header-path %t/enums-default.h
+// RUN: %FileCheck %s < %t/enums-default.h
+
 public enum E {
     case x(Double)
     case y(UnsafeRawPointer?)
@@ -23,6 +26,11 @@ public enum E {
     public func printSelf() {
         print("self")
     }
+}
+
+public enum E2 {
+    case foobar
+    case baz
 }
 
 public struct S {
@@ -166,6 +174,15 @@ public struct S {
 // CHECK-NEXT: #endif
 // CHECK-NEXT:     vwTable->initializeWithTake(destStorage, srcStorage, metadata._0);
 // CHECK-NEXT:   }
+
+// CHECK: class SWIFT_SYMBOL({{.*}}) E2 final {
+// CHECK: SWIFT_INLINE_THUNK operator cases() const {
+// CHECK: }
+// CHECK-NEXT: }
+// CHECK-EMPTY:
+// CHECK-NEXT: SWIFT_INLINE_THUNK swift::Int getHashValue() const SWIFT_SYMBOL({{.*}});
+// CHECK-NEXT: private:
+
 // CHECK:      namespace Enums SWIFT_PRIVATE_ATTR SWIFT_SYMBOL_MODULE("Enums") {
 // CHECK:        SWIFT_INLINE_THUNK E E::_impl_x::operator()(double val) const {
 // CHECK-NEXT:     auto result = E::_make();

